scripts/linters/pylint_extensions_test.py

```diff
@@ -3272,3 +3272,71 @@ def some_random_function(param1, param2):
 
         with self.checker_test_object.assertNoMessages():
             self.checker_test_object.checker.visit_functiondef(def_node)
+
+
+class DisallowDunderMetaclassCheckerTests(unittest.TestCase):
+
+    def test_wrong_metaclass_usage_raises_error(self):
+        checker_test_object = testutils.CheckerTestCase()
+        checker_test_object.CHECKER_CLASS = (
+            pylint_extensions.DisallowDunderMetaclassChecker)
+        checker_test_object.setup_method()
+
+        metaclass_node = astroid.extract_node(
+            """
+            class FakeClass(python_utils.OBJECT):
+                def __init__(self, fake_arg):
+                    self.fake_arg = fake_arg
+                def fake_method(self, name):
+                    yield (name, name)
+            class MyObject: #@
+                __metaclass__ = FakeClass
+                def __init__(self, fake_arg):
+                    self.fake_arg = fake_arg
+            """)
+
+        with checker_test_object.assertAddsMessages(
+            testutils.Message(
+                msg_id='no-dunder-metaclass',
```

Comment:
Can we have a test to assert the message generated by this check? (Currently, it just validates the id)

Reply:
All the tests in this file seem to test only the id.

Reply:
Right, but we've realised it's a good practice to also test the message string. And we want to introduce this practice slowly via code reviews, that's why you can do it too.

Reply:
@gp201 Right, but over time we've realised it's a good practice to also test the message string, and we are trying to enforce this slowly via code reviews. Your code will eventually serve as a good example for others to follow.

Reply:
Yeah, I agree with @sajalasati (And thanks for the explanation). I think if we had the test in the first place then we would have caught the following issues:
https://github.com/oppia/oppia/pull/12394#discussion_r609253001
https://github.com/oppia/oppia/pull/12394#discussion_r609596626

Reply:
As per offline discussion, we found that we can add a test for the message but it needs some minor structure changes, @gp201 will work on writing tests for messages in a separate PR. 

@gp201 Can you please create an issue for this task and assign it to yourself? (Once done, I'll approve and merge this PR! :) )

Reply:
@DubeySandeep PTAL https://github.com/oppia/oppia/issues/12507